File: helpers/pyband/pyband/coin.py
class Coin(object):
    def __init__(self, amount: int, denom: str) -> None:
        self.amount = amount
        self.denom = denom

    def __eq__(self, o: "Coin") -> bool:
        return self.amount == o.amount and self.denom == o.denom

    @classmethod
    def from_json(cls, coin) -> "Coin":
        return cls(int(coin["amount"]), coin["denom"])

    def as_json(self) -> dict:
        return {"amount": str(self.amount), "denom": self.denom}

    def validate(self) -> bool:
        if self.amount < 0:
            raise ValueError("Expect amount more than 0")

        if len(self.denom) == 0:
            raise ValueError("Expect denom")

        return True

File: helpers/bandchain2.js/src/coin.ts
export default class Coin {
  amount: number
  denom: string

  constructor(amount: number, denom: string) {
    this.amount = amount
    this.denom = denom
  }

  static fromJson(coin: { amount: number; denom: string }): Coin {
    return new Coin(coin.amount, coin.denom)
  }

  asJson(): { amount: string; denom: string } {
    return { amount: this.amount.toString(), denom: this.denom }
  }

  validate(): boolean {
    if (!Number.isInteger(this.amount)) throw Error('amount is not an integer')
    if (this.amount < 0) throw Error('Expect amount more than 0')
    if (this.denom.length === 0) throw Error('Expect denom')

    return true
  }
}
